packages/grafana-ui/src/components/Table/TableNG/hooks.ts

```diff
@@ -455,49 +391,24 @@ export function useRowHeight({
   hasNestedFrames,
   defaultHeight,
   expandedRows,
-  typographyCtx: { calcRowHeight, avgCharWidth },
+  typographyCtx,
 }: UseRowHeightOptions): number | ((row: TableRow) => number) {
-  const [wrappedColIdxs, hasWrappedCols] = useMemo(() => {
-    let hasWrappedCols = false;
-    return [
-      fields.map((field) => {
-        if (field.type !== FieldType.string) {
-          return false;
-        }
-
-        const cellOptions = getCellOptions(field);
-        const wrapText = 'wrapText' in cellOptions && cellOptions.wrapText;
-        const type = cellOptions.type;
-        const result = !!wrapText && type !== TableCellDisplayMode.Image;
-        if (result === true) {
-          hasWrappedCols = true;
-        }
-        return result;
-      }),
-      hasWrappedCols,
-    ];
-  }, [fields]);
+  const lineCounters = useMemo(() => buildRowLineCounters(fields, typographyCtx), [fields, typographyCtx]);
+  const hasWrappedCols = useMemo(() => lineCounters.some((lc) => lc !== null), [lineCounters]);
 
   const colWidths = useMemo(
     () => columnWidths.map((c) => c - 2 * TABLE.CELL_PADDING - TABLE.BORDER_RIGHT),
     [columnWidths]
   );
 
-  const maxWrapCellOptions = useMemo<GetMaxWrapCellOptions>(
-    () => ({
-      colWidths,
-      avgCharWidth,
-      wrappedColIdxs,
-    }),
-    [colWidths, avgCharWidth, wrappedColIdxs]
-  );
-
   const rowHeight = useMemo(() => {
     // row height is only complicated when there are nested frames or wrapped columns.
     if (!hasNestedFrames && !hasWrappedCols) {
       return defaultHeight;
     }
 
+    // this cache should get blown away on resize, data refresh, updated fields, etc.
+    const cache: number[] = [];
```

Comment:
one exciting change: we can cache the row height in the case where we do return a function. this means fewer calls to the calculation logic regardless of the implementation, which is a big win, because both RDG and us need to call the `rowHeight` method like 4 times per render of the table per row.